packages/server/src/queries/complex/orderbooks/spot-price.ts

```diff
@@ -0,0 +1,35 @@
+import { Chain } from "@osmosis-labs/types";
+import cachified, { CacheEntry } from "cachified";
+import { LRUCache } from "lru-cache";
+
+import { DEFAULT_LRU_OPTIONS } from "../../../utils/cache";
+import { queryOrderbookSpotPrice } from "../../osmosis";
+
+const orderbookSpotPriceCache = new LRUCache<string, CacheEntry>(
+  DEFAULT_LRU_OPTIONS
+);
+
+export function getOrderbookSpotPrice({
+  orderbookAddress,
+  chainList,
+  quoteAssetDenom,
+  baseAssetDenom,
+}: {
+  orderbookAddress: string;
+  quoteAssetDenom: string;
+  baseAssetDenom: string;
+  chainList: Chain[];
+}) {
+  return cachified({
+    cache: orderbookSpotPriceCache,
+    key: `orderbookSpotPrice-${orderbookAddress}-${quoteAssetDenom}-${baseAssetDenom}`,
+    ttl: 1000 * 60 * 2, // 2 minutes
+    getFreshValue: () =>
+      queryOrderbookSpotPrice({
+        orderbookAddress,
+        chainList,
+        quoteAssetDenom,
+        baseAssetDenom,
+      }).then(({ data }) => data.spot_price),
```

Comment:
NIT: Since this is in the complex folder, where we can make modifications / do processing on raw queries, what are your thoughts on adding the `Dec` return type here instead of in the tRPC procedure

Reply:
Sure